.infra/checkstyle/import-control.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE import-control PUBLIC "-//Checkstyle//DTD ImportControl Configuration 1.4//EN" "https://checkstyle.org/dtds/import_control_1_4.dtd">
+
+<import-control pkg="org.junitpioneer">
+
+	<allow pkg=".*" regex="true" />
```

Comment:
Had a quick look at the [docs](https://checkstyle.sourceforge.io/config_imports.html#ImportControl) (on SourceForge 🤭), do I get this right: within `org.junitpionner` (and its "subpackages"), allow everything (`.*`)? (As your commit message says, preparation...)

Reply:
In the current state of the project we don't _actually_ have optional dependencies. Once this PR gets merged, we can ask contributors (and maintainers?) that:
 - Features using the optional dependency be in their own package (?)
 - Add a line allowing packages from the optional dependency only in the feature-related package (?)
Did I get this right?

Reply:
Yeah, this is a neutral change that has no practical consequences, but foreshadows future additions. I 100% stole this from #492 because I'd prefer a git blame to point to this commit/PR/issue rather than the "random" extension that first needed it.